Cache `isModuleOutputEnabled` or `isModuleEnabled` status

sreichel opened the pull request:
### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

If `isModuleOutputEnabled` or `isModuleEnabled` is called multiple times for the same Module, get config once.

### Related Pull Requests
<!-- related pull request placeholder -->

1. see OpenMage/magento-lts#4320

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. add xdebug breapoints to `isModuleOutputEnabled()`
2. ...

```php
    var_dump($test->isModuleOutputEnabled('Mage_Catalog'));
    var_dump($test->isModuleOutputEnabled('Mage_Catalog'));
    var_dump($test->isModuleOutputEnabled('Mage_Catalog'));
    var_dump($test->isModuleOutputEnabled('a'));
    var_dump($test->isModuleOutputEnabled('a'));
    var_dump($test->isModuleOutputEnabled('b'));
```

kiatng commented on app/code/core/Mage/Core/Helper/Abstract.php:
```diff
@@ -42,6 +42,10 @@ abstract class Mage_Core_Helper_Abstract
      */
     protected $_layout;
 
+    protected array $modulesDisabled = [];
```
Traditionally, protected var has prefix '_'. Do we want to break tradition?

sreichel replied:
i'd not stick with old traditions.

i think we had PRs that added protected properties w/o underscore.

justinbeaty commented:
XML / DB config is already cached, how much time is this actually saving?

sreichel commented:
![xh1](https://github.com/user-attachments/assets/2de01f5a-094c-4d95-8453-64b1a6b74f1b)

For the example above,

sreichel commented:
> even though the "Disable Modules Output" can be configured at the store scope

I did not see that and did not test it. Thanks.

Config is alerady cached, so it brings no benfits. Check if module is enabled still save some calls. (seems even faster ... numbers from ddev + xhprof + xhgui)

sreichel commented:
https://github.com/OpenMage/magento-lts/pull/4320#discussion_r1821212246
> However, I only see two calls to this method

Updated:

- moved methods to parent abstract class ... so its also available for frontend block classes too
- added trait to provide same funtionality to models, resource,-models, ... (instead of adding helper methods multiple times)
